packages/beacon-node/src/sync/range/utils/hashBlocks.ts

```diff
@@ -1,14 +1,23 @@
-import {allForks} from "@lodestar/types";
+import {allForks, RootHex} from "@lodestar/types";
 import {IChainForkConfig} from "@lodestar/config";
-import {byteArrayConcat} from "../../../util/bytes.js";
+import {toHex} from "@lodestar/utils";
 
 /**
- * Hash SignedBeaconBlock in a byte form easy to compare only
- * @param blocks
- * @param config
+ * String to uniquely identify block segments. Used for peer scoring and to compare if batches are equivalent.
  */
-export function hashBlocks(blocks: allForks.SignedBeaconBlock[], config: IChainForkConfig): Uint8Array {
-  return byteArrayConcat(
-    blocks.map((block) => config.getForkTypes(block.message.slot).SignedBeaconBlock.hashTreeRoot(block))
-  );
+export function hashBlocks(blocks: allForks.SignedBeaconBlock[], config: IChainForkConfig): RootHex {
+  switch (blocks.length) {
+    case 0:
+      return "0x";
+    case 1:
+      return toHex(config.getForkTypes(blocks[0].message.slot).SignedBeaconBlock.hashTreeRoot(blocks[0]));
+    default: {
+      const block0 = blocks[0];
+      const blockN = blocks[blocks.length - 1];
```

Comment:
Is it possible to halt syncing with a batch like: `block ... bad block ... block`?

Reply:
This hash is only used for peer scoring, and attribute processing errors to the right set of peers